include/mbedtls/ssl.h

```diff
@@ -1724,7 +1724,15 @@ struct mbedtls_ssl_context {
     int MBEDTLS_PRIVATE(early_data_state);
 #endif
 
-    unsigned MBEDTLS_PRIVATE(badmac_seen);       /*!< records with a bad MAC received    */
+    /** Multipurpose field.
+     *
+     * - DTLS: records with a bad MAC received.
+     * - TLS: accumulated length of handshake fragments (up to in_hslen).
+     *
+     * This field is multipurpose in order to preserve the ABI in the
+     * Mbed TLS 3.6 LTS branch.
+     */
+    unsigned MBEDTLS_PRIVATE(badmac_seen);
```

Comment:
In my first iteration, I made minimal changes. But I think we should rename the field to something like `badmac_seen_or_in_hsfraglen`, to reduce confusion.